vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc

```diff
@@ -980,99 +1018,112 @@ void AdsImpl::CheckReadyAdServe(
     }
   }
 
-  auto category = GetWinnerOverTimeCategory();
-  ServeAdFromCategory(category);
+  auto categories = GetWinningCategories();
+  ServeAdFromCategories(categories);
 }
 
-void AdsImpl::ServeAdFromCategory(
-    const std::string& category) {
-  BLOG(INFO) << "Serving ad for category: " << category;
-
+void AdsImpl::ServeAdFromCategories(
+    const std::vector<std::string>& categories) {
   std::string catalog_id = bundle_->GetCatalogId();
   if (catalog_id.empty()) {
     FailedToServeAd("No ad catalog");
     return;
   }
 
-  if (!category.empty()) {
-    auto callback =
-        std::bind(&AdsImpl::OnServeAdFromCategory, this, _1, _2, _3);
-    ads_client_->GetAds(category, callback);
+  if (categories.empty()) {
+    BLOG(INFO) << "No categories";
+    ServeUntargetedAd();
     return;
   }
 
-  BLOG(INFO) << "Category is empty, trying again with untargeted category";
+  BLOG(INFO) << "Serving ad from categories:";
+  for (const auto& category : categories) {
+    BLOG(INFO) << "  " << category;
+  }
 
-  ServeUntargetedAd();
+  auto callback =
+      std::bind(&AdsImpl::OnServeAdFromCategories, this, _1, _2, _3);
+  ads_client_->GetAds(categories, callback);
 }
 
-void AdsImpl::OnServeAdFromCategory(
+void AdsImpl::OnServeAdFromCategories(
     const Result result,
-    const std::string& category,
+    const std::vector<std::string>& categories,
     const std::vector<AdInfo>& ads) {
   auto eligible_ads = GetEligibleAds(ads);
   if (!eligible_ads.empty()) {
-    ServeAd(category, eligible_ads);
+    BLOG(INFO) << "Found " << eligible_ads.size() << " eligible ads";
+    ServeAd(eligible_ads);
     return;
   }
 
-  if (ServeAdFromParentCategory(category, eligible_ads)) {
-    return;
+  BLOG(INFO) << "No eligible ads found in categories:";
+  for (const auto& category : categories) {
+    BLOG(INFO) << "  " << category;
   }
 
-  BLOG(INFO) << "No ads found in \"" << category
-      << "\" category, trying again with untargeted category";
+  if (ServeAdFromParentCategories(categories)) {
+    return;
+  }
 
   ServeUntargetedAd();
 }
 
-bool AdsImpl::ServeAdFromParentCategory(
-    const std::string& category,
-    const std::vector<AdInfo>& ads) {
-  auto pos = category.find_last_of('-');
-  if (pos == std::string::npos) {
-    return false;
-  }
+bool AdsImpl::ServeAdFromParentCategories(
+    const std::vector<std::string>& categories) {
+  std::vector<std::string> parent_categories;
+  for (const auto& category : categories) {
+    auto pos = category.find_last_of('-');
+    if (pos == std::string::npos) {
```

Comment:
nit: would suggest a const for `'-'` if a token/delimiter.

Reply:
Fixed